caches/infinispan/src/main/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProvider.java

```diff
@@ -63,6 +68,8 @@
 
     private String nfsBaseDir;
 
+    // use weak key map to avoid the memory occupy for long time of the transfer
+    //FIXME: should we mark this as static to make it used across all instances(for lock also)?
```

Comment:
I think we can allow the application to manage the instances in the JVM...for instance, Indy uses initializes a single instance and uses a CDI provider to make it available. It's conceivable that we might want to use two of these in some weird system in the future. As long as they don't store references to the same file, it should work. OTOH, maybe there's something we should do to define a "root" within which this provider has responsibility for everything. I'm not sure how we could do that elegantly, TBH.